src/components/dialogs/network-modifications/tabular-modification/tabular-modification-form.tsx

```diff
@@ -0,0 +1,199 @@
+        disabled: !csvColumns,
+        resetTrigger: typeChanged,
+    });
+
+    const postProcessFile = useCallback(
+        (fileData: any) => {
+            fileData = fileData.map((row: any) => {
+                let modification: Modification = {};
+                Object.getOwnPropertyNames(row).forEach((key) => {
+                    modification[key] = row[key];
+                });
+                return modification;
+            });
+            setValue(MODIFICATIONS_TABLE, fileData, { shouldDirty: true });
+        },
+        [setValue]
+    );
+
+    const watchTable = useWatch({
+        name: MODIFICATIONS_TABLE,
+    });
+
+    useEffect(() => {
+        if (selectedFileError) {
+            setValue(MODIFICATIONS_TABLE, []);
+        } else if (selectedFile) {
+            Papa.parse(selectedFile as any, {
+                header: true,
+                skipEmptyLines: true,
+                complete: function (results) {
+                    postProcessFile(results.data);
+                },
+            });
+        }
+    }, [postProcessFile, selectedFile, selectedFileError, setValue]);
+
+    const typesOptions = useMemo(() => {
+        const equipmentTypesToExclude = new Set([
+            EQUIPMENT_TYPES.SWITCH,
+            EQUIPMENT_TYPES.LCC_CONVERTER_STATION,
+            EQUIPMENT_TYPES.VSC_CONVERTER_STATION,
+            EQUIPMENT_TYPES.HVDC_CONVERTER_STATION,
+            EQUIPMENT_TYPES.STATIC_VAR_COMPENSATOR,
+            EQUIPMENT_TYPES.DANGLING_LINE,
+            EQUIPMENT_TYPES.THREE_WINDINGS_TRANSFORMER,
+            EQUIPMENT_TYPES.HVDC_LINE,
+            EQUIPMENT_TYPES.SHUNT_COMPENSATOR,
+            EQUIPMENT_TYPES.SUBSTATION,
+            EQUIPMENT_TYPES.VOLTAGE_LEVEL,
+            EQUIPMENT_TYPES.TWO_WINDINGS_TRANSFORMER,
+            EQUIPMENT_TYPES.LINE,
+            EQUIPMENT_TYPES.BATTERY,
+        ]);
+        return Object.values(EQUIPMENT_TYPES).filter(
+            (equipmentType) => !equipmentTypesToExclude.has(equipmentType)
+        );
+    }, []);
+
+    const handleChange = useCallback(() => {
+        setTypeChanged(!typeChanged);
+        setValue(MODIFICATIONS_TABLE, []);
+    }, [setValue, typeChanged]);
+
+    const equipmentTypeField = (
+        <AutocompleteInput
+            isOptionEqualToValue={richTypeEquals}
+            name={TYPE}
+            label="Type"
+            options={typesOptions}
+            onChangeCallback={handleChange}
+            getOptionLabel={richTypeLabel}
+            size={'small'}
+            formProps={{ variant: 'filled' }}
+        />
+    );
+
+    const defaultColDef = useMemo(
+        () => ({
+            flex: 1,
+            filter: true,
+            sortable: true,
+            resizable: false,
+            lockPinned: true,
+            wrapHeaderText: true,
+            autoHeaderHeight: true,
+            cellRenderer: DefaultCellRenderer,
+        }),
+        []
+    );
+    interface ColumnDefinition {
```

Comment:
it is not necessary to create this interface, you can use directly "ColDef" from 'ag-grid-community' module